builtin/credential/aws/path_role_test.go

```diff
@@ -986,6 +988,87 @@ func TestAwsVersion(t *testing.T) {
 	}
 }
 
+// This test was used to reproduce https://github.com/hashicorp/vault/issues/7418
+// and verify its fix.
+// Please run it at least 3 times to ensure that passing tests are due to actually
+// passing, rather than the region being randomly chosen tying to the one in the
+// test through luck.
+func TestRoleResolutionWithSTSEndpointConfigured(t *testing.T) {
+	if enabled := os.Getenv("VAULT_ACC"); enabled == "" {
+		t.Skip()
+	}
+
+	// ex. "arn:aws:iam::123456789012:role/MyRole"
```

Comment:
I would suggest a comment to help other developers in the future know how to determine what to set this to, not just what syntax it should follow. Something like:

```suggestion
	/* ARN of an AWS role that Vault can query during testing.
	   This role should exist in your current AWS account and your credentials
	   should have iam:GetRole permissions to query it.
	 */
```